connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -901,6 +947,19 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
                 if (request != null && started) {
                     updateListener.onRestartRequest(request);
                 }
+            } else if (record.key().startsWith(TASK_COUNT_RECORD_PREFIX)) {
```

Comment:
We're adding this `else if` clause to a method that's now ~250 lines long. I think we can factor the block of each `if` and `else if` into its own method.

Reply:
Yeah, we've definitely reached that point 👍

While doing this decomposition I kept the bodies for each new method identical to the `if`/`else if` blocks that they were extracted from, with these exceptions:
- Log messages that include the record key are adjusted to use the connector name in its place (this doesn't drop any information)
- Calls to `Object::getClass` for logging messages are all converted to calls to the newly-introduced and null-safe `className` method
- The `@SuppressWarnings("unchecked")` annotation is removed from method signatures and is instead added only to assignments within the method bodies that require it